src/core/src/op/acos.cpp

```diff
@@ -28,53 +35,26 @@ std::shared_ptr<ov::Node> ov::op::v0::Acos::clone_with_new_inputs(const OutputVe
     return std::make_shared<Acos>(new_args.at(0));
 }
 
-OPENVINO_SUPPRESS_DEPRECATED_START
-namespace acosop {
-namespace {
-template <ov::element::Type_t ET>
-inline bool evaluate(const ngraph::HostTensorPtr& arg0, const ngraph::HostTensorPtr& out, const size_t count) {
-    using T = typename ov::element_type_traits<ET>::value_type;
-    ngraph::runtime::reference::acos<T>(arg0->get_data_ptr<ET>(), out->get_data_ptr<ET>(), count);
-    return true;
-}
-
-bool evaluate_acos(const ov::HostTensorPtr& arg0, const ov::HostTensorPtr& out, const size_t count) {
-    bool rc = true;
-    out->set_unary(arg0);
```

Comment:
In the original evaluate, shape and element type of the output `HostTensor` were set.
Don't we need corresponding code to set the shape and element type of the output `ov::Tensor`?

https://github.com/openvinotoolkit/openvino/blob/679369c7071d624a334a2c52145cc0457a56e318/src/core/src/runtime/host_tensor.cpp#L151-L154
cc: @ilyachur

Reply:
I'm not sure if this is really required for ov::Tensor @ilyachur can comment about it?

Reply:
I wouldn't want to add such functionality to `ov::Tensor` I think the current functionality should be enough.
Maybe we can add `set_element_type()` to tensor

Reply:
Yes we can use set_shape. Also as I see right now `ov::Tensor` doesn't have `set_element_type()` method. We can add it if it is needed.

Reply:
The `set_shape` added. I think it can be required in case of constant folding (when lower/upper bounds of shape are evaluated) then the output can have dynamic shape and output tensor has to be set the correct shape.

Reply:
Great, I've approved the changes, the need of `set_element_type` can be considered separately.